apps/experiments/models.py

```diff
@@ -739,7 +739,8 @@ def create_new_version(self, version_description: str | None = None, make_defaul
     def get_fields_to_exclude(self):
         return super().get_fields_to_exclude() + ["is_default_version", "public_id", "version_description"]
 
-    def compare_with_latest(self):
+    @cached_property
+    def is_dirty(self) -> bool:
```

Comment:
My issue with caching it with redis is that we'd have to update the cache everytime we make a change to a related model which can become too laborious, but in principle, yes that would have been better. The core issue that was raised is that it is not always very obvious when there are new changes. Maybe we should instead let the user click a button to check for changes? Or maybe we simply rename the "create new version" button to "check for changes". It takes you to the same view and all, but maybe this would make it the obvious way to check for changes. Thoughts?

Also, I don't know how useful the create new version button in the experiment edit view is. I've always found it easier and better suited to use the button on the versions tab. Don't know if you have strong feelings about this?

Reply:
I think it's fine to keep the create version button on the edit page for now. 

In terms of checking the dirty state, my main concern is that it will make the page slow to load. We could make it load async using htmx. As long as there are no changes we don't have to cache it but as soon as there are changes we cache.

Reply:
> As long as there are no changes we don't have to cache it but as soon as there are changes we cache.

This doesn't solve the issue of us having to update the cache everytime we make a change to a related model?

Reply:
I think our options are either 1. go the htmx route, but without redis caching or 2. revert for now

Reply:
> > As long as there are no changes we don't have to cache it but as soon as there are changes we cache.
> 
> This doesn't solve the issue of us having to update the cache everytime we make a change to a related model?

It does because once it's dirty it remains dirty so we don't need to keep checking. And we only need to bust the cache when a new version is created.

Reply:
Reverted